lib/toggl/TogglService.ts

```diff
@@ -135,6 +141,13 @@ export default class TogglService {
       this.noticeAPINotAvailable();
     }
     apiStatusStore.set(this._ApiAvailable);
+
+    if (interval > 0) {
```

Comment:
Re: your comment:

> Merged the code paths, but now refreshApiConnection sets an interval just to destroy it one cycle later. Should work, but doesn't feel like the best way to do it.

You're right, the right primitive to use here might be `useTimeout`. Since there's technically no deadline for the `refreshApiConnection` promise to resolve, an interval might start another refresh promise while the previous is still resolving. Setting a timeout for the next refresh at the end of the current one guarantees this cannot happen.

Reply:
Just to clarify, do you mean `setTimeout`, or a different library?
I'm also not sure if `setTimeout` will be harder to lifecycle manage (whether the loop will automatically get cleaned up when unloading the plugin).

Reply:
`setTimeout` is indeed what I meant. I figured that the Obsidian API would have a similar primitive for cleaning up those like has for `setInterval`, but apparently not. A pity.

What solution would you propose to work with?

Reply:
New behavior I'm noticing---now when there's a connection error when refreshing, the refresh function errors before starting a new interval. Refreshing then stops until manually refreshing, at which it starts again. Not the behavior we want!

Perhaps refresh logic should check for the existence of an auto-refresh interval, and create if it doesn't exist?